### PR TITLE
[MM-48003] Work Templates: add telemetry

### PR DESCRIPTION
**Pull request by JulienTant**
#### Summary
Add telemetry except for all the ones in "After creation" of this document: https://www.figma.com/file/xS2v61Rlq2DiLG8QF3fIPU/MMO-MVP?node-id=668%3A174151&t=jggkxeunzSTKNin3-0

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-48003

#### Related Pull Requests
N/A

#### Screenshots
N/A

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: components/work_templates/components/menu.tsx [hunk 1]
```diff
@@ -63,6 +64,10 @@ interface MenuProps {
 const Menu = ({className, disableQuickUse, categories, workTemplates, currentCategoryId, changeCategory, onTemplateSelected}: MenuProps) => {
     const {formatMessage} = useIntl();
 
+    useEffect(() => {
+        trackEvent('work_templates', 'pageview_menu');
```
Comment (pvev) [inline components/work_templates/components/menu.tsx:68]: I'd suggest to put the category in a constant value https://github.com/mattermost/mattermost-webapp/blob/master/utils/constants.tsx#L729

File: components/work_templates/index.tsx [hunk 2]
```diff
@@ -88,6 +89,10 @@ const WorkTemplateModal = () => {
     const teamId = useSelector(getCurrentTeamId);
     const playbookTemplates = useSelector((state: GlobalState) => state.entities.worktemplates.playbookTemplates);
 
+    useEffect(() => {
+        trackEvent('work_templates', 'open_modal');
```
Comment (pvev) [inline components/work_templates/index.tsx:93]: is the value `open_modal` significant enough to keep track of the intention, I mean, is not too generic?

File: components/work_templates/index.tsx [hunk 3]
```diff
@@ -190,9 +209,22 @@ const WorkTemplateModal = () => {
         closeModal();
     };
 
+    const trackAction = (action: string, actionFn: () => void) => {
```
Comment (pvev) [inline components/work_templates/index.tsx:212]: 👍

**Review by pvev — APPROVED** [review]
LGTM! thanks @JulienTant

**Review by mhd-sln — APPROVED** [review]
Awesome! Thanks @JulienTant! :shipit:

File: components/work_templates/index.tsx (hunk 2 above)
Comment (JulienTant) [reply components/work_templates/index.tsx:93]: I added it as a way to measure feature usage, because the modal opening = someone uses the feature.

**Pull request merged**